connection_test.go

```diff
@@ -970,3 +970,76 @@ func TestPeerStatusChangeServer(t *testing.T) {
 	})
 	assert.Len(t, changes, 0, "unexpected peer status changes")
 }
+
+func TestContextCanceledOnTCPClose(t *testing.T) {
+	// 1. Context canceled warning is expected as part of this test
+	// add log filter to ignore this error
+	// 2. We use our own relay in this test, so disable the relay
+	// that comes with the test server
+	opts := testutils.NewOpts().NoRelay().AddLogFilter("simpleHandler OnError", 1)
+
+	testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
+		serverDoneC := make(chan struct{})
+		callForwarded := make(chan struct{})
+
+		ts.RegisterFunc("test", func(ctx context.Context, args *raw.Args) (*raw.Res, error) {
+			defer close(serverDoneC)
+			assert.Equal(t, "client", CurrentCall(ctx).CallerName(), "wrong caller name")
+			close(callForwarded)
+			<-ctx.Done()
+			assert.Equal(t, "context canceled", ctx.Err().Error(), "ctx.Err() returned unexpected error")
+			return &raw.Res{}, nil
+		})
+
+		// Set up a relay that can be used to terminate conns
+		// on both sides i.e. client and server
+		relayFunc := func(outgoing bool, f *Frame) *Frame {
+			return f
+		}
+		relayHostPort, shutdown := testutils.FrameRelay(t, ts.HostPort(), relayFunc)
+
+		// Make a call with a long timeout. We shutdown the relay
+		// immediately after the server receives the call. Expected
+		// behavior is for both client/server to be done with the call
+		// immediately after relay shutsdown
+		ctx, cancel := NewContext(20 * time.Second)
+		defer cancel()
+
+		clientCh := ts.NewClient(&testutils.ChannelOpts{
+			ServiceName: "client",
```

Comment:
does the client name matter for this test? if not, just do `clientCh := ts.NewClient(nil)`